search/TopologicalSearch.cpp

```diff
@@ -26,4 +37,24 @@ bool TopologicalSearch::isInCircle(const Coordinates &center, Meters radius, con
     return inCircle;
 }
 
+bool TopologicalSearch::isInCircle(const Coordinates &center, Meters radius, const PointList &entityGeometry) const {
+    std::vector<Coordinates> coords = entityGeometry.getPoints();
+    auto coordsIter = coords.begin();
+    Coordinates prevCoord = *coordsIter;
+    for (coordsIter = next(coordsIter); coordsIter < coords.end(); coordsIter++) {
```

Comment:
Why did you use next(coordsIter)? Doesn't it skip the first segment?

Reply:
The line is between prevCoord and coord, so coordsIter does start from the second coordinate, but it shouldn't skip the first line.